src/components/reviews/ReviewsBreakdown.jsx

```diff
@@ -4,41 +4,115 @@ import API_KEY from '../../config.js';
 import StarRating from '../shared/StarRating.jsx';
 import BreakdownBar from './BreakdownBar.jsx';
+import CharacteristicBar from './CharacteristicBar.jsx';
 
 class ReviewsBreakdown extends React.Component {
   constructor(props) {
     super(props);
+    this.state = {
+      ratingsToFilter: {}
+    };
+    this.toggleRatingFilter = this.toggleRatingFilter.bind(this);
+  }
+
+  clearAllFilters() {
+    this.setState({ ratingsToFilter: {} });
+    this.props.setFilter((review) => true);
+  }
+
+  toggleRatingFilter(stars) {
+    const { ratingsToFilter } = this.state;
+    if (ratingsToFilter[stars]) {
+      ratingsToFilter[stars] = false;
+    } else {
+      ratingsToFilter[stars] = true;
+    }
+    let allFiltersOff = true;
+    for (let rating in ratingsToFilter) {
+      if (ratingsToFilter[rating]) {
+        allFiltersOff = false;
+      }
+    }
+    this.props.setFilter((review) => {
+      return (allFiltersOff || ratingsToFilter[review.rating]);
+    });
+    this.setState({ ratingsToFilter });
   }
 
   render() {
     const { reviewsMetaData } = this.props;
+    const { ratingsToFilter } = this.state;
     if (reviewsMetaData) {
-      const { ratings, recommended } = reviewsMetaData;
+      const { ratings, recommended, characteristics } = reviewsMetaData;
       const ratingsArray = [];
+      const filteredRatingsArray = [];
       let ratingsSum = 0;
       let reviewCount = 0;
       for (let rating in ratings) {
         ratingsSum += parseInt(ratings[rating]) * rating;
         reviewCount += parseInt(ratings[rating]);
         ratingsArray.push(rating);
+        if (ratingsToFilter[rating]) {
+          filteredRatingsArray.push(rating);
+        }
       }
       ratingsArray.reverse();
       if (reviewCount === 0) {
         return null;
       }
       const avgRating = Math.round(ratingsSum / reviewCount * 10) / 10;
       const recommendPercentage = Math.round((recommended.true / reviewCount) * 100);
+
+      const characteristicsNames = [];
+      for (let name in characteristics) {
+        characteristicsNames.push(name);
+      }
+      const characteristicsText = {
+        'Size': ['A size too small', 'A size too wide'],
+        'Width': ['Too narrow', 'Too wide'],
+        'Comfort': ['Uncomfortable', 'Perfect'],
+        'Quality': ['Poor', 'Perfect'],
+        'Length': ['Runs Short', 'Runs long'],
+        'Fit': ['Runs tight', 'Runs long']
+      };
       return (
         <div className='reviewsBreakdown'>
           <span className='avgRating'>{avgRating}</span>
           <StarRating rating={avgRating} />
           <br></br>
-          {ratingsArray.map((stars) => (
-            <BreakdownBar
+          {filteredRatingsArray.length > 0 ? (
+            <>
+              <span>Filters: </span>
+              <span className='filterSpan' onClick={() => this.clearAllFilters()}>
+                Clear all
+              </span>
+              <br></br>
+            </>
+          ) : null}
```

Comment:
You might consider using the logical conditional rendering(&&) when you want to display null for the false case.